.golangci.yml

```diff
@@ -18,8 +18,19 @@ linters:
     - ineffassign
     - misspell
     - unconvert
+    - typecheck
```

Comment:
took the liberty to align these with upstream with the two exceptions below which would cause lots of warnings

Reply:
Would it be ok then to merge https://github.com/ava-labs/coreth/pull/764 before? 🤔 
It cleans up linter configuration etc. and only adds new linters for **our** files. Then we can add linters in this PR on top of it? I didn't take it out of draft because I was originally waiting for a libevm x master merge first 😄